context-free.md

```diff
@@ -2,15 +2,19 @@
 
 ## What is easy to do in this language? Why is it easy?
 
+Context Free is a Domain-Specific Language whose limited-expressiveness in terms of its methods and parameters makes it ideal for generating and creating complex patterns and images. It contains parameters such as chirality, transformation, rotation, flipping which create domain focus on image creation since these concepts are all easily-applicable and mostly-specific to images. At the same time, these are concepts comprehensible by most regardless of their experience in graphic design thus increasing the usability of the language to the general public and anybody willing and able to learn it. Its limited-expressiveness in terms of the fact that it defines images only through certain primitive shapes and a few shape and colour readjustment methods means that it can easily be learnt and used by beginners. Generating detailed and patterned images such as fractals is also much easier and more convenient to do using this language as compared to using a typical photo editor.
 
 ## What is hard to do in this language? Why is it hard?
 
+One difficulty in using ContextFree is that its limitation to only using squares, triangles, and circles means that images such as sketches of people, animals, etc. - anything requiring the usage of irregular shapes - are near impossible to produce with accuracy: its usage remains limited to geometric shapes and patterns. In addition to this, when using ContextFree for image creation instead of a photo-editor or a pen and paper, one cannot see the results of their actions and methods (e.g. skewing, flipping, rotating, etc.) in real time and must wait until they have "rendered" the program to do so. Since the designs being created are often complex, this makes it difficult, especially for beginners, to be able to predict the output of a particular program. Were the image able to reflect the changes in the program in real-time, usage of ContextFree may have been much more convenient. Another thing I found difficult to do in this language was debugging. While the program is able to point out the line where the error lies, there are no specifics provided as to the nature of the error thus making it difficult, especially for beginners, to fix their programs.
```

Comment:
I also find it hard to generate graphs without certain patterns in it. And I agree with the debugging problem because it also causes me a lot of trouble. I tried to change some variables but it rendered out the whole black and it took me a long time to see where the problem was. However, I do not think "rendered" function will cause trouble for beginners because they can press the button any time they want so they know what is the next step. In my opinion, it will be hard to add the real-time rendered graph due to the large usage of memory.